core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadWriteLDAPUserStoreManager.java

```diff
@@ -913,11 +914,10 @@ public void doSetUserClaimValues(String userName, Map<String, String> claims, St
                             userAttributeSeparator = claimSeparator;
                         }
                         if (claimEntry.getValue().contains(userAttributeSeparator)) {
-                            StringTokenizer st = new StringTokenizer(claimEntry.getValue(), userAttributeSeparator);
-                            while (st.hasMoreElements()) {
-                                String newVal = st.nextElement().toString();
+                            String[] claimValues = claimEntry.getValue().split(Pattern.quote(userAttributeSeparator));
+                            for (String newVal : claimValues) {
```

Comment:
Can we name "newVal" variable as "claimValue". This is due to the fact that the variable name specified here do not interpret its use